ksp_plugin_adapter/flight_planner.cs

```diff
@@ -139,6 +139,14 @@ class FlightPlanner : VesselSupervisedWindowRenderer {
 
   private void RenderFlightPlan(string vessel_guid) {
     using (new UnityEngine.GUILayout.VerticalScope()) {
+      if (UnityEngine.GUILayout.Button("Rebase")) {
+        var status = plugin.FlightPlanRebase(
+            vessel_guid, predicted_vessel.GetTotalMass());
+        UpdateStatus(status, null);
+        if (status.ok()) {
+          UpdateVesselAndBurnEditors();
```

Comment:
This seems wrong.  You cannot call this function during `Repaint`.  Look at the existing call, it happens at `Layout` time.

I suspect that you can just drop lines 146-148 and immediately return from this function.  The right thing should happen at the next cycle.  Of course, this needs to be tested in game.